traffic_ops/traffic_ops_golang/user/current.go

```diff
@@ -80,3 +144,184 @@ WHERE u.id=$1
 	u.LocalUser = util.BoolPtr(localPassword.Valid)
 	return u, nil
 }
+
+func ReplaceCurrent(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var userRequest tc.CurrentUserUpdateRequest
+	if err := json.NewDecoder(r.Body).Decode(&userRequest); err != nil {
+		errCode = http.StatusBadRequest
+		userErr = fmt.Errorf("Couldn't parse request: %v", err)
+		api.HandleErr(w, r, tx, errCode, userErr, nil)
+		return
+	}
+
+	user, exists, err := dbhelpers.GetUserByID(inf.User.ID, tx)
+	if err != nil {
+		sysErr = fmt.Errorf("Getting user by ID %d: %v", inf.User.ID, err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+	if !exists {
+		sysErr = fmt.Errorf("Current user (#%d) doesn't exist... ??", inf.User.ID)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	if err := userRequest.User.UnmarshalAndValidate(&user); err != nil {
+		errCode = http.StatusBadRequest
+		userErr = fmt.Errorf("Couldn't parse request: %v", err)
+		api.HandleErr(w, r, tx, errCode, userErr, nil)
+		return
+	}
+
+	changePasswd := false
+
+	// obfuscate passwords (UnmarshalAndValidate checks for equality with ConfirmLocalPassword)
+	// TODO: check for valid password via bad password list like Perl did? User creation doesn't...
```

Comment:
I don't believe this is true. The shared handlers call the `Validate` method after decoding the json request body, and if the localPasswd is not nil, it checks to make sure the username and localPassword are a good login pair.

Reply:
Users doesn't use GenericCreate, the only validation I can see being done before the user is `INSERT`ed is checking that the password is non-nil. I also can't find a `tc.User.Validate` function, so if there is more validation being done my question is: where? I'd like to see it so I know how to write that enforcement.

Reply:
`POST /users` uses `api.CreateHandler`, which eventually calls `TOUser.Validate`.

Reply:
Ah thanks. I'm used to seeing `Validate` in `/lib/go-tc/`. Should be checking password validity now.